Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -6,12 +6,104 @@
  *
  */
 
-#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
+#include <QApplication>
+#include <QtGui/private/qguiapplication_p.h>
+#include <QShortcutEvent>
+#include <QTimer>
+
+#include <AzQtComponents/Components/WindowDecorationWrapper.h>
 
+#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
 #include <AzToolsFramework/ActionManager/Menu/MenuManagerInterface.h>
 
 namespace AzToolsFramework
 {
+    //! This class is used to install an event filter on each widget that the action contexts are registered to in order to properly handle
+    //! ambiguous shorcuts.
+    class ActionContextWidgetWatcher : public QObject
+    {
+    public:
+        explicit ActionContextWidgetWatcher(QObject* parent = nullptr)
+            : QObject(parent)
+        {
+        }
+
+        bool eventFilter(QObject* watched, QEvent* event) override
+        {
+            switch (event->type())
+            {
+            case QEvent::Shortcut:
+            {
+                QShortcutEvent* shortcutEvent = static_cast<QShortcutEvent*>(event);
+                if (shortcutEvent->isAmbiguous())
+                {
+                    QWidget* watchedWidget = qobject_cast<QWidget*>(watched);
+                    for (QAction* action : watchedWidget->actions())
+                    {
+                        if (action->shortcut() == shortcutEvent->key())
+                        {
+                            action->trigger();
+                            return true;
+                        }
+                    }
+                }
+                break;
+            }
+            case QEvent::ShortcutOverride:
+            {
+                QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+                int keyInt = keyEvent->key();
+                Qt::KeyboardModifiers modifiers = keyEvent->modifiers();
+                if (modifiers & Qt::ShiftModifier)
+                {
+                    keyInt += Qt::SHIFT;
+                }
+                if (modifiers & Qt::ControlModifier)
+                {
+                    keyInt += Qt::CTRL;
+                }
+                if (modifiers & Qt::AltModifier)
+                {
+                    keyInt += Qt::ALT;
+                }
+                if (modifiers & Qt::MetaModifier)
+                {
+                    keyInt += Qt::META;
+                }
+
+                QKeySequence keySequence(keyInt);
+
+                // FIXME: Originally it looked like we could check this shortcutMap to determine if the ShortcutOverride
+                // was ambiguous, but unfortunately that doesn't seem to be the case :/
+                auto globalShortcutMap = &QGuiApplicationPrivate::instance()->shortcutMap;
+                bool isAmbiguous = globalShortcutMap->hasShortcutForKeySequence(keySequence);
+                if (isAmbiguous)
+                {
```

Comment:
If you need this information for actions defined in the Action Manager... 

The Action Manager holds a pointer to all QActions in the corresponding EditorAction internal class instance, so it should be possible to either retrieve this information or create a cache somewhere pretty easily.

The `HotKeyMapping` class in the HotKeyManager also holds a map between shortcut and action identifier. We may have to create a map per Action Context instead of having all in one place like it is right now, but that may also be a simple way to bypass this issue.

Of course it's harder in the current situation where we're also taking actions that are not defined in the Action Manager into account.

Reply:
Yeah it would be nice to just lookup in our own internal cache, but certainly made trickier with our current environment where we have actions registered outside of the Action Manager. I'll keep digging.